transaction-scenarios/src/scenarios/non_fungible_resource.rs

```diff
@@ -0,0 +1,405 @@
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-unfreeze-deposit",
+                        |builder| builder.unfreeze_burn(state.vault1.unwrap()),
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-recall-unfreezed-vault",
+                        |builder| {
+                            builder
+                                .recall_non_fungibles(state.vault1.unwrap(), btreeset!(NonFungibleLocalId::integer(130)))
+                                .try_deposit_batch_or_abort(config.user_account_1.address)
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .finalize(|core, config, state| {
+                Ok(ScenarioOutput {
+                    interesting_addresses: DescribedAddresses::new()
+                    .add("user_account_1", config.user_account_1.address.clone())
+                    .add("user_account_2", config.user_account_2.address.clone())
+                    .add(
+                        "integer_non_fungible_resource",
+                        state.integer_non_fungible_resource.unwrap(),
+                    )
+                    .add(
+                        "string_non_fungible_resource",
+                        state.string_non_fungible_resource.unwrap(),
+                    )
+                    .add(
+                        "bytes_non_fungible_resource",
+                        state.bytes_non_fungible_resource.unwrap(),
+                    )
+                    .add(
+                        "ruid_non_fungible_resource",
+                        state.ruid_non_fungible_resource.unwrap(),
+                    ),
+                })
+            })
+    }
+}
+
+#[derive(ScryptoSbor, ManifestSbor)]
+struct ComplexFungibleData {
+    a: u32,
+    b: Vec<String>,
+    c: AnotherObject,
+}
+
+#[derive(ScryptoSbor, ManifestSbor)]
+struct AnotherObject {
+    f1: BTreeMap<String, (u8, (u16, Vec<Vec<u8>>))>,
+}
+
+impl NonFungibleData for ComplexFungibleData {
+    const MUTABLE_FIELDS: &'static [&'static str] = &["a", "c"];
```

Comment:
Might be cool to have a transaction which updates the NF data

Reply:
And also - because of the tombstoning - a transaction which burns a Non-Fungible, and attempts to recreate it?

Reply:
Although we can possibly make a list of these various suggestions and get someone to implement these

Reply:
(Have added these ideas to the slack thread, so feel free to ignore them for now)

Reply:
Cool. I did consider this when writing the scenarios. Will revisit this when I get some downtime. :)